src/report/sagas.test.js

```diff
@@ -0,0 +1,107 @@
+import LoggingService from '@edx/frontend-logging';
+import { takeEvery } from 'redux-saga/effects';
+import { runSaga } from 'redux-saga';
+import * as ApiService from './service';
+import saga, { handleFetchReports } from './sagas';
+
+jest.mock('@edx/frontend-logging');
+
+beforeEach(() => {
+  LoggingService.mockClear();
+});
+
+describe('handleFetchReports', () => {
+  it('correctly handles a non-empty API response', async () => {
```

Comment:
I realize that there is some code duplication between `correctly handles a non-empty API response` and `correctly handles an empty API response`, but because of the differences in the mocked response, I couldn't really substantially refactor this common code out into a `beforeEach` block, for example.

Reply:
Can't we just put the three `const` variables: `dispatched`, `programKey` and `mock` to be in the scope of `describe` function?